### PR TITLE
Tokens fail to retry due to empty nonce

### PR DESCRIPTION
**Pull request by tmancey**
Fixes https://github.com/brave/brave-browser/issues/7222
Requires: https://github.com/brave/brave-core/pull/4106

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:

See https://github.com/brave/brave-browser/issues/7222

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: vendor/bat-native-confirmations/src/bat/confirmations/internal/refill_tokens.cc [hunk 1]
```diff
@@ -53,12 +53,19 @@ void RefillTokens::Refill(
 
   public_key_ = public_key;
 
+  nonce_ = "";
```
Comment (emerick) [inline vendor/bat-native-confirmations/src/bat/confirmations/internal/refill_tokens.cc]: Seems unnecessary in the constructor, but fine by me if you'd just like to be explicit.

File: vendor/bat-native-confirmations/src/bat/confirmations/internal/refill_tokens.cc [hunk 2]
```diff
@@ -53,12 +53,19 @@ void RefillTokens::Refill(
...
+
   RequestSignedTokens();
 }
 
 void RefillTokens::RetryGettingSignedTokens() {
   BLOG(INFO) << "Retry getting signed tokens";
 
+  if (nonce_.empty()) {
```
Comment (emerick) [inline vendor/bat-native-confirmations/src/bat/confirmations/internal/refill_tokens.cc]: Is there any danger of endlessly retriggering this?

Comment (tmancey) [reply vendor/bat-native-confirmations/src/bat/confirmations/internal/refill_tokens.cc]: There is, however right now we have no retry count so I will raise a ticket

File: vendor/bat-native-confirmations/src/bat/confirmations/internal/refill_tokens.cc (hunk 1 above)
Comment (tmancey) [reply vendor/bat-native-confirmations/src/bat/confirmations/internal/refill_tokens.cc]: We need the nonce to reset if starting without the need to retry

File: vendor/bat-native-confirmations/src/bat/confirmations/internal/refill_tokens.cc (hunk 2 above)
Comment (emerick) [reply vendor/bat-native-confirmations/src/bat/confirmations/internal/refill_tokens.cc]: OK, good idea.

File: vendor/bat-native-confirmations/src/bat/confirmations/internal/refill_tokens.cc (hunk 1 above)
Comment (emerick) [reply vendor/bat-native-confirmations/src/bat/confirmations/internal/refill_tokens.cc]: Got it, I misread this as being the class's constructor. Sorry!

File: vendor/bat-native-confirmations/src/bat/confirmations/internal/refill_tokens.cc (hunk 2 above)
Comment (tmancey) [reply vendor/bat-native-confirmations/src/bat/confirmations/internal/refill_tokens.cc]: https://github.com/brave/brave-browser/issues/7226

**Comment by tmancey** [comment]
CI failed with unrelated error (network audit)

**Pull request merged**